# HG changeset patch
# User Sergi Almacellas Abellana <sergi@koolpi.com>
Use PYSON in colors attributes of view definition

issue5301
review15981002

Index: trytond/trytond/modules/account/move.py
===================================================================

--- a/trytond/trytond/modules/account/move.py
+++ b/trytond/trytond/modules/account/move.py
@@ -18,7 +18,7 @@
     Button
 from trytond.report import Report
 from trytond import backend
-from trytond.pyson import Eval, Bool, PYSONEncoder
+from trytond.pyson import Eval, Bool, If, PYSONEncoder
 from trytond.transaction import Transaction
 from trytond.pool import Pool, PoolMeta
 from trytond.rpc import RPC
@@ -1439,9 +1439,11 @@
             if not journal.view:
                 return result

+            colors = PYSONEncoder().encode(If(Eval('state', '') == 'draft',
+                    'red', 'black'))
             xml = '<?xml version="1.0"?>\n' \
                 '<tree string="%s" editable="top" on_write="on_write" ' \
-                'colors="red:state==\'draft\'">\n' % title
+                'colors=\'%s\'>\n' % (title, colors)
             fields = set()
             for column in journal.view.columns:
                 fields.add(column.field.name)

